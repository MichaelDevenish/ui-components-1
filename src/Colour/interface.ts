export interface ColourProps {
  description: string
}

export interface SProps extends ColourProps {
  hex: string
}

export interface ColourObject {
  name: string;
  hex: string;
}

export interface ColourWithNameAndValue {
  [key: string]: string
}
