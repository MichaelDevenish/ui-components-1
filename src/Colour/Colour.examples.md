### A Colour Component
```jsx
const { brandColours } = require('./Colour');

<Colour hex={brandColours.intelliRoyalBlue} description="intelli-royal-blue" />
```
