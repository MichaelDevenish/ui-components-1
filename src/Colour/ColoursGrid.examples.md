### Brand Colours
```jsx
const { brandColoursArray } = require('./Colour');

<ColoursGrid colours={brandColoursArray} />
```

### App Colours
```jsx
const { appColoursArray } = require('./Colour');

<ColoursGrid colours={appColoursArray} />
```