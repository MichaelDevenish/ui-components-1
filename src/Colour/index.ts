export {
  Colour,
  brandColours,
  appColours
} from './Colour'

export {
  ColoursGrid
} from './ColoursGrid'
