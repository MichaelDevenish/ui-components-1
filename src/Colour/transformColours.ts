import { kebabCase } from 'lodash'

export const transformColours = (c: any) => Object.keys(c).map(k => ({
  name: kebabCase(k),
  hex: c[k]
}))
