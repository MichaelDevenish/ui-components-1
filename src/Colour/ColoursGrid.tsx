import React from 'react'
import styled from 'styled-components'
import { ColourObject, Colour, ColourWithNameAndValue } from './Colour'

export interface ColourGridProps {
  colours: ColourObject[]
}

export class ColoursGrid extends React.PureComponent<ColourGridProps> {
  render () {
    const WrappedList = styled.div`
      display: flex;
      flex-wrap: wrap;
      max-width: 50rem;
      margin-left: 0;
    `
    const { colours } = this.props

    return (
      <WrappedList>
        {colours.map(c => <Colour hex={c.hex} description={c.name} key={c.name} />)}
      </WrappedList>
    )
  }
}
