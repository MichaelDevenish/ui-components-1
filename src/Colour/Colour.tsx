import React, { Fragment } from 'react'
import styled from 'styled-components'
import { transformColours } from './transformColours'

export interface ColourProps {
  description: string
}

export interface SProps extends ColourProps {
  hex: string
}

export interface ColourObject {
  name: string
  hex: string
}

export interface ColourWithNameAndValue {
  [key: string]: string
}

export const brandColours: ColourWithNameAndValue = {
  intelliRoyalBlue: '#432DF3',
  intelliGreen: '#04DBAC',
  intelliYellow: '#FFDD37',
  intelliRed: '#FF4D50',
  intelliCyan: '#03DEEB',
  intelliPurple: '#A35AFF',
  intelliCharcoal: '#303035',
  intelliGrey1: '#B3B3B5',
  intelliGrey2: '#E4E4E5',
  intelliBaseWhite: '#F5F7F9',
  intelliOffWhite: '#F5F5F5'
}

export const appColours: ColourWithNameAndValue = {
  intelliAppBlue: '#0F68FA',
  intelliDarkGrey: '#47525D',
  intelliGreyBlue1: '#C3CCD0',
  intelliGreyBlue2: '#DFE5E8',
  intelliWhite: '#FFFFFF',
  intelliRoyalBlueLight: '#EBE9FE',
  intelliAppBlueLight: '#D7F2FF',
  intelliGreenLight: '#E0FEF8',
  intelliYellowLight: '#FFF9DA',
  intelliRedLight: '#FFE6E6'
}

/**
 * This component is for demonstration of the colour plates purpose
 */
export class Colour extends React.PureComponent<SProps> {
  render () {
    const { hex, description } = this.props
    const Box = styled.div`
      width: 10em;
      height: 7.5em;
      background-color: ${hex};
      margin-bottom: 1em;
    `
    const ColourName = styled.p`
      font-weight: 600;
    `
    return (<div>
      <Box />
      <ColourName> {description} </ColourName>
      <p> {hex} </p>
    </div>)
  }
}

export const brandColoursArray: ColourObject[] = transformColours(brandColours)
export const appColoursArray: ColourObject[] = transformColours(appColours)
