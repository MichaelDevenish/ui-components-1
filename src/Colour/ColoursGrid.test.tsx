import { mount } from 'enzyme'
import React from 'react'
import { ColoursGrid } from './'
import { brandColoursArray } from './Colour'

describe('<ColoursGrid />', () => {
  it(`should render a colour component`, () => {
    const wrapper = mount(
      <ColoursGrid colours={brandColoursArray} />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
