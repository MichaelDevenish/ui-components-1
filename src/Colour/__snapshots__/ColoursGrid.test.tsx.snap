// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColoursGrid /> should render a colour component 1`] = `
<ColoursGrid
  colours={
    Array [
      Object {
        "hex": "#432DF3",
        "name": "intelli-royal-blue",
      },
      Object {
        "hex": "#04DBAC",
        "name": "intelli-green",
      },
      Object {
        "hex": "#FFDD37",
        "name": "intelli-yellow",
      },
      Object {
        "hex": "#FF4D50",
        "name": "intelli-red",
      },
      Object {
        "hex": "#03DEEB",
        "name": "intelli-cyan",
      },
      Object {
        "hex": "#A35AFF",
        "name": "intelli-purple",
      },
      Object {
        "hex": "#303035",
        "name": "intelli-charcoal",
      },
      Object {
        "hex": "#B3B3B5",
        "name": "intelli-grey-1",
      },
      Object {
        "hex": "#E4E4E5",
        "name": "intelli-grey-2",
      },
      Object {
        "hex": "#F5F7F9",
        "name": "intelli-base-white",
      },
      Object {
        "hex": "#F5F5F5",
        "name": "intelli-off-white",
      },
    ]
  }
>
  <styled.div>
    <div
      className="sc-bdVaJa etdbDy"
    >
      <Colour
        description="intelli-royal-blue"
        hex="#432DF3"
        key="intelli-royal-blue"
      >
        <div>
          <styled.div>
            <div
              className="sc-bwzfXH cWLzbN"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-htpNat daVmCj"
            >
               
              intelli-royal-blue
               
            </p>
          </styled.p>
          <p>
             
            #432DF3
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-green"
        hex="#04DBAC"
        key="intelli-green"
      >
        <div>
          <styled.div>
            <div
              className="sc-bxivhb hSmeNb"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-ifAKCX fDiTuQ"
            >
               
              intelli-green
               
            </p>
          </styled.p>
          <p>
             
            #04DBAC
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-yellow"
        hex="#FFDD37"
        key="intelli-yellow"
      >
        <div>
          <styled.div>
            <div
              className="sc-EHOje jLhXoF"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-bZQynM bydMxI"
            >
               
              intelli-yellow
               
            </p>
          </styled.p>
          <p>
             
            #FFDD37
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-red"
        hex="#FF4D50"
        key="intelli-red"
      >
        <div>
          <styled.div>
            <div
              className="sc-gzVnrw RNCrr"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-htoDjs hEDuLJ"
            >
               
              intelli-red
               
            </p>
          </styled.p>
          <p>
             
            #FF4D50
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-cyan"
        hex="#03DEEB"
        key="intelli-cyan"
      >
        <div>
          <styled.div>
            <div
              className="sc-dnqmqq jSpDwW"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-iwsKbI jbxArO"
            >
               
              intelli-cyan
               
            </p>
          </styled.p>
          <p>
             
            #03DEEB
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-purple"
        hex="#A35AFF"
        key="intelli-purple"
      >
        <div>
          <styled.div>
            <div
              className="sc-gZMcBi eRHIIL"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-gqjmRU bupxyl"
            >
               
              intelli-purple
               
            </p>
          </styled.p>
          <p>
             
            #A35AFF
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-charcoal"
        hex="#303035"
        key="intelli-charcoal"
      >
        <div>
          <styled.div>
            <div
              className="sc-VigVT iLopOi"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-jTzLTM biUQWd"
            >
               
              intelli-charcoal
               
            </p>
          </styled.p>
          <p>
             
            #303035
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-grey-1"
        hex="#B3B3B5"
        key="intelli-grey-1"
      >
        <div>
          <styled.div>
            <div
              className="sc-fjdhpX klIbXh"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-jzJRlG fIiydF"
            >
               
              intelli-grey-1
               
            </p>
          </styled.p>
          <p>
             
            #B3B3B5
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-grey-2"
        hex="#E4E4E5"
        key="intelli-grey-2"
      >
        <div>
          <styled.div>
            <div
              className="sc-cSHVUG fehlNa"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-kAzzGY GyKas"
            >
               
              intelli-grey-2
               
            </p>
          </styled.p>
          <p>
             
            #E4E4E5
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-base-white"
        hex="#F5F7F9"
        key="intelli-base-white"
      >
        <div>
          <styled.div>
            <div
              className="sc-chPdSV gBUpVU"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-kgoBCf bzkTnp"
            >
               
              intelli-base-white
               
            </p>
          </styled.p>
          <p>
             
            #F5F7F9
             
          </p>
        </div>
      </Colour>
      <Colour
        description="intelli-off-white"
        hex="#F5F5F5"
        key="intelli-off-white"
      >
        <div>
          <styled.div>
            <div
              className="sc-kGXeez jQSpWt"
            />
          </styled.div>
          <styled.p>
            <p
              className="sc-kpOJdX gBQQgN"
            >
               
              intelli-off-white
               
            </p>
          </styled.p>
          <p>
             
            #F5F5F5
             
          </p>
        </div>
      </Colour>
    </div>
  </styled.div>
</ColoursGrid>
`;
